docs/docs/deploying-to-digitalocean-droplet.md

```diff
@@ -0,0 +1,230 @@
+
+Change the root directory configuration of Nginx in the default server block file
+
+1. Go to /etc/nginx/sites-available/
+
+   ```bash
+   cd /etc/nginx/sites-available/
+   ```
+
+2. Open the file default
+
+   ```bash
+   sudo vim default
+   ```
+
+3. Edit the file and make the following changes for below-mentioned fields, leave the rest of the fields as it is.
+
+   ```bash
+   server {
+     root <path to my-gatsby-app>/public;
+
+     index index.html index.htm index.nginx-debian.html;
+
+     server_name <your-domain-name>;
+
+     location / {
+       try_files $uri $uri/ =404;
+     }
+   }
+   ```
+
+4. Restart the ngnix service,
+
+   ```bash
+   sudo systemctl restart nginx
+   ```
+
+5. Configure your domain to point to the IP address of your droplet. Go to the Advanced DNS settings in your domain provider's console and put an A record that points to the IP address of the droplet.
+
+6. By this time, you can view your site live at `<your-domain>`.
+
+### Configuring SSL for your Gatsby Site.
+
+Follow the below steps to configure your site with a free SSL certificate from Lets Encrypt using their Certbot cli tool.
+
+However, it's advised to use a premium SSL certificate from any SSL Certificate provider.
+
+1. Install Certbot onto your droplet.
+
+   You'll need to add the Certbot PPA to your list of repositories. To do so, run the following commands,
+
+   ```bash
+   sudo apt-get update
+   sudo add-apt-repository ppa:certbot/certbot
+   sudo apt-get update
+   ```
+
+   Run the following commands to install certbot.
+
+   ```bash
+   sudo apt-get install certbot python3-certbot-nginx
+   ```
+
+2. Generate the certificate. Certbot will automatically edit and configure the ngnix config file for the SSL and point to the certificate file.
+
+   Run the following command,
+
+   ```bash
+   sudo certbot --nginx
+   ```
+
+3. If you are using Certbot for the first time on this droplet then you will be prompted to enter your e-mail for recovery purpose.
+
+4. Agree to the license agreement on prompt.
+
+   `Note`: You will be prompted to select the domain for which you want to generate the certificate. Select the domain configured in [this](#configuring-ngnix-to-point-to-your-gatsby-site's-`public`-directory-and-add-your-domain) step.
+
+   `Note`: You will be prompted to choose the option to redirect HTTP requests to HTTPS, you may choose on your needs.
+   (Suggested, to choose to redirect HTTP to HTTPS)
+
+5. Restart the nginx service.
+
+   ```bash
+   sudo systemctl restart nginx
+   ```
+
+6. Now, you can access your site at `<your-domain>` with a secure connection.
+
+### View your Gatsby site live
+
+Once you've followed along with all the steps and configuration properly, you can visit your site live at `<your-domain>`.
+
+Whenever there's an update to your site, run a `sudo gatsby build` in the root of your `<my-gatsby-app>` and your changes will be live.
```

Comment:
We could achieve CI or automated builds with the use of Github Actions and Git hooks? But don't you think it would make this guide more of a tutorial? We could keep the guide pretty raw and basic, we can obviously make a tutorial on building CI for this deployment option and link it to this guide. What are your views?

Reply:
Hey there. I think at least mentioning it would be useful–even a link to an offsite resource for automating builds. I don't think that would tip a guide toward a tutorial, as this is already pretty sequential even with standalone headings as opposed to "steps". But we could get this one merged and open that as a separate issue, perhaps.

Reply:
Hey, I have searched for some good off-site resources, but I don't find any good resource to add as a link in this doc. Those resources are not exact to what we need. I think adding them, would mislead people. Instead, I'm planning to write a micro-tutorial on building automated CI for Gatsby and Digitalocean on some off-site platform which we can add as a link to this doc later on. Or else, as you suggested we can open this as a separate issue to write a tutorial. Let me know what do you think.

Reply:
I think opening a separate issue to follow this PR is a good idea, that way we can track it!

Reply:
Yes. Perfect! 😃